src/engine/GroupBy.cpp

```diff
@@ -1094,6 +1097,34 @@ template <typename A>
 concept SupportedAggregates =
     ad_utility::isTypeContainedIn<A, GroupBy::Aggregations>;
 
+// _____________________________________________________________________________
+// Visitor function to extract values from the result of an evaluation of
+// the child expression of an aggregate, and subsequently processing the values
+// by calling the `increment` function of the corresponding aggregate.
+auto makeProcessGroupsVisitor =
```

Comment:
Global variables that are only present in the .cpp file should either be
`static` (can even be `static constexpr` in this case)
or inside an anonymous
`namespace {  /* your function here */}`

to avoid strange bugs if there somewhere is another function with the same name.